lib/externalApis/tenderdash/BlockchainListener.js

```diff
@@ -76,37 +98,56 @@ class BlockchainListener extends EventEmitter {
    * @param {number} [timeout] - timeout to reject after
    * @return {Promise<Object>}
    */
-  waitForTransaction(hashString, timeout = 60000) {
+  waitForTransactionToBeProvable(hashString, timeout = 60000) {
     const topic = BlockchainListener.getTransactionEventName(hashString);
-    let handler;
+    const txInBlockTopic = BlockchainListener
+      .getTransactionAddedToTheBlockEventName(hashString.toLowerCase());
+
+
+    let txHandler;
+    const txPromise = new Promise(((resolve) => {
+      txHandler = this.createPromiseHandler(topic, resolve);
```

Comment:
Maybe we move `createPromiseHandler` here? It calls only from here and I think the code will be understandable since it's not clear what exact handler with method supposed to create.